File: back/internal/domain/telemetry.go
package domain

type Telemetry struct {
	TelemetryID    int64
	DeviceID       int64
	RecordedAt     int64
	MetricType     string
	MetricValue    float64
	Unit           string
	RoomName       string
	LocationName   string
	BatteryLevel   int64
	SignalStrength int64
}

type TelemetryIngest struct {
	DeviceID       int64
	RecordedAt     int64
	HasRecordedAt  bool
	MetricType     string
	MetricValue    float64
	Unit           string
	RoomName       string
	LocationName   string
	BatteryLevel   int64
	HasBattery     bool
	SignalStrength int64
	HasSignal      bool
}

type IngestResult struct {
	Telemetry Telemetry
	Alerts    []Alert
}

File: client/src/entities/telemetry.ts
export type Telemetry = {
  telemetryId: number;
  deviceId: number;
  recordedAt: number;
  metricType: string;
  metricValue: number;
  unit: string;
  roomName: string;
  locationName: string;
  batteryLevel: number;
  signalStrength: number;
};

export type TelemetryListResponse = {
  telemetry: Telemetry[];
};

export type TelemetryIngestRequest = {
  deviceId: number;
  recordedAt?: number;
  metricType: string;
  metricValue: number;
  unit: string;
  roomName: string;
  locationName: string;
  batteryLevel?: number;
  signalStrength?: number;
};

export type TelemetryMqttRequest = {
  topic: string;
  payload: TelemetryIngestRequest;
};

export type TelemetryIngestResponse = {
  telemetry: Telemetry;
  alerts: {
    alertId: number;
    locationId: number;
    deviceId: number;
    createdAt: number;
    severity: string;
    message: string;
    isResolved: boolean;
    resolvedAt: number;
  }[];
};

export type TelemetryQuery = {
  metricType?: string;
  from?: number;
  to?: number;
  limit?: number;
};
